src/GovUk.Education.ExploreEducationStatistics.Common.Tests/Extensions/DateTimeOffsetTestExtensions.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions;
+
+public static class DateTimeOffsetTestExtensions
+{
+    /// <summary>
+    /// Strip the nanosecond component from a DateTimeOffset. Helpful for testing in scenarios where precision can
+    /// change throughout a DateTimeOffset's lifecycle.
+    ///
+    /// An example would be when comparing DateTimeOffsets when one has been retrieved from PostgreSQL, where the
+    /// precision is less than that of C# itself.
+    /// 
+    /// </summary>
+    /// <param name="offset"></param>
+    /// <returns></returns>
+    public static DateTimeOffset TruncateNanoseconds(this DateTimeOffset offset)
+    {
+        return new DateTimeOffset(
+            offset.Year, 
+            offset.Month, 
+            offset.Day, 
+            offset.Hour, 
+            offset.Minute, 
+            offset.Second,
+            offset.Millisecond, 
+            offset.Microsecond, 
+            TimeSpan.Zero);
+    }
```

Comment:
Ahh cool, I didn't realise this had been tackled already.  Happy to go with the already-established approach, although this has the minute advantage that it is more accurate, only removing the difference of the nanoseconds, plus you only have to apply it to the "before" date.  E.g:

Assert.Equal(before.TruncateNanoseconds(), after);

rather than:

Assert.Equal(before.ToUnixTimeSeconds(), after.ToUnixTimeSeconds());

I suppose we could also have used the 3rd `precision` argument if we wanted to ignore the nanosecond difference?

Reply:
Yeah the first time I came across this problem I used the `precision` argument until I realised that we were using `ToUnixTimeSeconds` elsewhere.

We could add a new Assert.Equal extension method to `AssertExtensions` that uses a default value of `precision`, so it doesn't rely on remembering and then specifying what the value of `precision` should be in each assertion, e.g. `TimeSpan.FromMicroseconds(0.1))`. I'm not sure what name I'd give that method.

I think either `TruncateNanoseconds` or this new extension method is the way to go because it's then more accurate than `ToUnixTimeSeconds` like you say.

Then it's just a question of when we update all the existing cases already using `ToUnixTimeSeconds`.

Reply:
I will throw it out to the team, thanks!

Reply:
We've decided on using the TruncateNanoseconds() approach, so I've updated all tests that used ToUnixTimeSeconds to use this instead.  Also added tests for the extension method.